pulp_rpm/app/metadata_parsing.py

```diff
@@ -173,7 +180,37 @@ def pkgcb(pkg):
             pkg(preaterepo_c.Package): a parsed metadata for a package
 
         """
+        nonlocal pkgid_warning_triggered
+        nonlocal nevra_warning_triggered
+
+        ERR_MSG = _(
+            "The repository metadata being synced into Pulp is erroneous in a way that "
+            "makes it ambiguous, and therefore we do not allow it to be synced in "
+            "'mirror_complete' mode. Please choose a sync policy which does not mirror "
+            "repository metadata.\n\n"
+            "Please read https://github.com/pulp/pulp_rpm/issues/2402 for more details."
+        )
+        WARN_MSG = _(
+            "The repository metadata being synced into Pulp is erroneous in a way that "
+            "makes it ambiguous (duplicate {}). Yum, DNF and Pulp try to handle these problems, "
+            "but unexpected things may happen.\n\n"
+            "Please read https://github.com/pulp/pulp_rpm/issues/2402 for more details."
```

Comment:
Given the issue I just discovered yesterday (https://github.com/rpm-software-management/createrepo_c/issues/305#issuecomment-1039268986) I kind of thing this message should be even stronger. 

The metadata they sync into the DB will actually be incorrect, DNF will probably handle it OK, but it'll be stuck like that until we do a migration.  And there's no way to parse it without it being incorrect.

But I don't know that we should entirely reject those repos in all modes because DNF doesn't, and we don't know how many there are.